src/ui/components/Library/Team/View/NewTestRuns/hooks/useTestRuns.ts

```diff
@@ -27,7 +28,16 @@ export function useTestRuns(): { testRuns: TestRun[]; status: Status } {
     teamId
   );
 
-  const testRuns = useMemo(() => [...value].reverse(), [value]);
+  useEffect(() => {
+    if (status === STATUS_PENDING) {
+      // To avoid resetting the test runs to an empty array when loading
```

Comment:
Seems like we should just be using _suspense_ here for this. It automatically gives you an _is pending_ signal which you can use to e.g. dim the list while you're loading new data (which we should do here btw)

Reply:
My understanding of Suspense is, where Suspense is used it needs to have promise inside of it and while this promise is pending Suspense is will fallback to "fallback" component. Please correct me if I'm wrong here.

Now this useTestRuns hook is used at Context level. Even if we use .read here that Suspense is signal is not available when using the value from context deep in child level component. If I'm right so far, maybe we can refactor to NOT use testRuns via Context but it may become little painful passing testRuns value to different components.

Reply:
You're already working with a Suspense cache. (That's what `testRunsIntervalCache` is.)

> While this promise is pending, Suspense will fall back to a 'fallback' component. Please correct me if I'm wrong.

That's only true if you're mounting something initially or if you suspend during a high priority update. [That's where `useTransition` comes in.](https://react.dev/reference/react/useTransition)

This way of using status to ignore pending updates feels kind of like you're trying to approximate what `useTransition` and Suspense give you already.